cczoo/tensorflow-serving-cluster/tensorflow-serving/docker/tf_serving_ccp/README.md

```diff
@@ -1,188 +1,224 @@
+![img](./Images/ps1.png "Technology Architecture")
 
-```shell
-git clone https://github.com/intelconfidential-computing-zoo.git
+- **client**: The client machine will encrypt the trained model and the TLS certificate used to establish a secure connection, and upload the tf-sering image after using ccp-cli pack to the CCP SGX client environment. At the same time, the client machine will also deploy the key to provide services, which are mainly used to authenticate the SGX side to ensure the integrity of the TensorFlow Serving inference service application running on the cloud and the feasibility of the SGX environment. After verification, the key is sent to the TensorFlow Serving inference service running in CCP SGX.
+
+- **CCP SGX**: The CCP SGX side provides the SGX confidential computing environment, and the TensorFlow Serving inference service runs in the SGX environment. When the inference service starts, a remote authentication request is sent to the client to obtain the secret model's key. After the verification is successful, the inference service will get the key sent by the client. So far, the inference service running in the SGX environment is running successfully and waiting for remote access requests.
+
+- **remote requester**: Third-party consumers send data to inference services running in the SGX confidential computing environment through secure transmission over the network. After the inference is completed, the return result is obtained.
+
+  *Note：In this practice, the client and the remote requester are deployed on the same machine, and the CCP SGX end is deployed on another machine.*
+
+## Prerequisites
+
+- Docker container engine: In order to facilitate the deployment of inference, this practice runs the three participants in a Docker container, and runs the inference service through Docker commands.
+- Ubuntu docker base images 20.04
+- TensorFlow 2.4.2 
+
+Reference for the environment information for this practice to run:
+- Specifications: encrypted memory ≥ 8G
+- Public IP
+- CCP software stack
+
+
+## Build the tf-serving image
+
+Users can build tf-serving images in the following two ways:
+
+### Download tf-serving image
+
+```
+sudo docker pull intelcczoo/tensorflow_serving:ccp_tf_serving_latest
+docker tag intelcczoo/tensorflow_serving:ccp_tf_serving_latest tf_serving:latest
 ```
 
+### Build your own tf-serving image
 
-- 1.1 Download the Model
+1. Download the TensorFlow Serving script code used in this practice and install the required packages.
 
-    First, use `download_model.sh` to download the pre-trained model file. It will
-    generate the directory `models/resnet50-v15-fp32` in current directory:
+    ```shell
+    git clone https://github.com/intel/confidential-computing-zoo.git
+    ```
 
+2. Download the model
     ```shell
-    cd <tensorflow-serving-cluster dir>/tensorflow-serving/docker/client
+    cd <tensorflow-serving-cluster dir>/tensorflow-serving/docker/clien
```

Comment:
cd <tensorflow-serving-cluster dir>/tensorflow-serving/docker/clien
 should be client?